### PR TITLE
feat(v3-preset): Add typescript support

### PR DESCRIPTION
**Pull request by ElijahKotyluk**
TS Support for Vuetify 3 CLI Alpha testing.

### To test

- Clone and check out this branch

In a separate directory
```bash
vue create my-app
```

#### Vue prompt
Manually select Vue 3, TypeScript, Vuex and Vue Router.
```bash
? Please pick a preset: (Use arrow keys)
    Default ([Vue 2] babel, eslint) 
    Default (Vue 3 Preview) ([Vue 3] babel, eslint) 
  ❯ Manually select features 
```

#### cd into app dir
```bash
cd my-app
```

#### Install vuetify/vue-cli-plugins
```bash
yarn add -D '../local/path/to/vue-cli-plugins-vuetify'
```

#### Invoke vuetify
```bash
vue invoke vuetify
```

#### Vuetify prompt
```bash
? Choose a preset: (Use arrow keys)
  Default (recommended) 
  Preview (v3 + vite)
  Prototype (rapid development) 
❯  V3 (alpha) 
  Configure (advanced) 
```

#### Run app
```bash
yarn dev
```

### CODE REVIEW

**Review by johnleider — CHANGES_REQUESTED** [review]
Seems like we have a mix of semicolons and no semicolons for line-endings. Is this intentional or something we can normalize? I'd prefer none if possible.

**Comment by ElijahKotyluk** [comment]
In the template component files?

File: packages/vue-cli-plugin-vuetify/generator/tools/vuetify.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ function renderFiles (api, { opts }) {
 }
 
 function addImports (api) {
-  api.injectImports(api.entryFile, "import vuetify from './plugins/vuetify';")
+  api.injectImports(api.entryFile, "import vuetify from './plugins/vuetify'")
```
Comment (johnleider) [inline packages/vue-cli-plugin-vuetify/generator/tools/vuetify.js:64]: linter should be catching this and changing the double quotes to single.

**Review by johnleider — CHANGES_REQUESTED** [review]

File: packages/vue-cli-plugin-vuetify/generator/tools/vuetify.js (hunk 1 above)
Comment (ElijahKotyluk) [reply packages/vue-cli-plugin-vuetify/generator/tools/vuetify.js:64]: Do you want it like this `'import vuetify from \'./plugins/vuetify\''`

Comment (johnleider) [reply packages/vue-cli-plugin-vuetify/generator/tools/vuetify.js:64]: disregard, I missed the inner quotations

**Review by johnleider — APPROVED** [review]

**Comment by glen-84** [comment]
I'm using webpack 5, and I'm seeing errors similar to this:

```sh
ERROR in ../../node_modules/vuetify/lib/components/VIcon/VIcon.mjs 11:0-61
Module not found: Error: Can't resolve '../../util' in 'D:\Programming\Projects\*\v4\front-end\node_modules\vuetify\lib\components\VIcon'
Did you mean 'index.mjs'?
BREAKING CHANGE: The request '../../util' failed to resolve only because it was resolved as fully specified
(probably because the origin is a '*.mjs' file or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
resolve '../../util' in 'D:\Programming\Projects\*\v4\front-end\node_modules\vuetify\lib\components\VIcon'
  using description file: D:\Programming\Projects\*\v4\front-end\node_modules\vuetify\package.json (relative path: ./lib/components/VIcon)
    Field 'browser' doesn't contain a valid alias configuration
    using description file: D:\Programming\Projects\*\v4\front-end\node_modules\vuetify\package.json (relative path: ./lib/util)
      Field 'browser' doesn't contain a valid alias configuration
      D:\Programming\Projects\*\v4\front-end\node_modules\vuetify\lib\util is not a file
 @ ../../node_modules/vuetify/lib/components/VIcon/index.mjs 1:0-47 1:0-47
 @ ../../node_modules/vuetify/lib/components/index.mjs 36:0-34 36:0-34
 @ ./vuetify.ts 5:0-53 10:4-14
 @ ./main.ts 4:0-32 5:31-38
```

[This](https://github.com/webpack/webpack/issues/11467#issuecomment-691873586) can be used as a workaround, but I believe that the correct fix is to make changes like this:

```diff
-import { convertToUnit, flattenFragments } from "../../util";
+import { convertToUnit, flattenFragments } from "../../util/index.mjs";
```

... which, I suppose, would be the following in the TSX files:

```diff
-import { convertToUnit, flattenFragments } from '@/util'
+import { convertToUnit, flattenFragments } from '@/util/index.mjs'
```

Could you please make these changes?

**Comment by ElijahKotyluk** [comment]
@glen-84 I'm going to look at supporting Webpack 5 as a separate issue.

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/App.js.vue [hunk 2]
```diff
@@ -27,5 +27,5 @@ export default {
   data: () => ({
     //
   }),
-};
+}
 </script>
```
Comment (glen-84) [inline packages/vue-cli-plugin-vuetify/generator/templates/v3/src/App.js.vue:31]: Add EOF newline.

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/App.ts.vue [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  },
+
+  <%_ } _%>
+  data () {
+    return {
+      //
+    }
+  },
+})
+</script>
```
Comment (glen-84) [inline packages/vue-cli-plugin-vuetify/generator/templates/v3/src/App.ts.vue:34]: Add EOF newline.

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/shims-vuetify.d.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+declare module 'vuetify'
+declare module 'vuetify/lib/components'
+declare module 'vuetify/lib/directives'
```
Comment (glen-84) [inline packages/vue-cli-plugin-vuetify/generator/templates/v3/src/shims-vuetify.d.ts:3]: Add EOF newline.

**Review by glen-84 — CHANGES_REQUESTED** [review]
This repo needs an EditorConfig file. 🙂

File: packages/vue-cli-plugin-vuetify/generator/templates/v3/src/shims-vuetify.d.ts (hunk 4 above)
Comment (ElijahKotyluk) [reply packages/vue-cli-plugin-vuetify/generator/templates/v3/src/shims-vuetify.d.ts:3]: Yes, it really does.  Thanks for catching that.

**Pull request merged**